game-app/game-core/src/main/java/tools/image/FileHelper.java

```diff
@@ -0,0 +1,21 @@
+package tools.image;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class FileHelper {
+
+  public static Path getFileInMapRoot(
```

Comment:
Would you mind adding a javadoc to this method? Just to help a maintainer understand this more quickly:
- what is meant by map root?
- what happens if hte file is not found?

We should also consider creating a 'MapFolder' object to own methods like this.

Reply:
> We should also consider creating a 'MapFolder' object to own methods like this.

Yeah, a lot of the code in the different classes is mostly copy-paste which is a bad smell. I'm sure if we were to extract the code to dedicated methods we can save like 50% of the code

Reply:
In case you're not convinced by this statement, just check the usages of this newly created method. Even though I already extracted a lot of common functionality the calling code still looks pretty much the same on the calling end.

If you have the time you might be able to come up with a creative idea to unify a lot of this code, but I'll just focus on this PR series for now

Reply:
Some thought would need to go into creating an object model that finally encapsulates a lot of these concepts. I think we might have a good chunk of that for example with "downloadedMap", it's perhaps the same object and just needs to be broadened.

The primary ask here though is for java-doc'ing please to clarify this method. It almost seems like it should be in java-extras but then again it uses TripleA specific terms which could use defining. If anything, the javadoc would allow a maintainer to read the docs rather than reverse engineering the code, which is useful for reducing maintenance cost.